game/src/engine/pickup/PickupService.h

```diff
@@ -0,0 +1,120 @@
+                                     const std::string& member);
+    void LoadAmmoCooldownInformation(const rapidjson::Value& ammo_object,
+                                     const std::string& member);
+    void LoadAmmoRespawnTimeInformation(const rapidjson::Value& ammo_object,
+                                        const std::string& member);
+
+    // Only for buckshot as it has other information as well.
+    void LoadAmmoOtherInformation(const rapidjson::Value& ammo_object,
+                                  const std::string& member);
+
+    /******** POWERUPS ******/
+    void LoadPowerupInformation(const rapidjson::Document& doc);
+
+    void LoadPowerupDurationInformation(const rapidjson::Value& powerup_object,
+                                        const std::string& member);
+
+    void LoadPowerupOtherInformation(const rapidjson::Value& powerup_object,
+                                     const std::string& member);
+    void LoadPowerupRespawnTimeInformation(
+        const rapidjson::Value& powerup_object, const std::string& member);
+
+    /****** GETTERS SETTERS *****/
+    std::array<std::string, 6> GetAmmoPickupNames();
+    float GetAmmoDamage(std::string ammo_type);
+    float GetAmmoDuration(std::string ammo_type);
+    float GetAmmoCooldown(std::string ammo_type);
+    float GetAmmoRespawnTime(std::string ammo_type);
+    float GetBuckshotAdditionalDetail(std::string detail_type);
+
+    std::array<std::string, 5> GetPowerupPickupNames();
+    float GetPowerupDuration(std::string powerup_type);
+    float GetPowerupMaxSpeeds(std::string powerup_type);
+    float GetPowerupRespawnTime(std::string powerup_type);
+
+    /********* EXECUTING AND HANDLING THE POWERUPS ************/
+
+    // Handle disabling the powerup
+    void HandleDisablingPowerup();
+
+    // Add the entity with the powerup they are holding
+    void AddEntityWithPowerup(Entity* entity, const std::string& powerup);
+
+    // Add the timers for the powerup when they are executed.
+    void AddEntityWithTimer(Entity* entity, const float& timer);
+
+    // To tell if the car should have the slowing down multiplier to them or not
+    bool IsVehicleSlowDown(Entity* entity);
+
+    // To tell if the car should have the slowing down multiplier to them or not
+    bool IsVehicleDisableHandling(Entity* entity);
+
+  private:
+    /******** AMMO ********/
+    std::unordered_map<std::string, float> ammo_damages_;
+    std::unordered_map<std::string, float> ammo_durations_;
+    std::unordered_map<std::string, float> ammo_cooldowns_;
+    std::unordered_map<std::string, float> buckshot_additional_details_;
+    std::unordered_map<std::string, float> ammo_respawn_times_;
+
+    /******** POWERUP ***/
+    std::unordered_map<std::string, float> powerup_durations_;
+    std::unordered_map<std::string, float> powerup_max_speeds_;
+    std::unordered_map<std::string, float> powerup_respawn_times_;
```

Comment:
Not sure if it's making this change at this point cause it's already done, but thought I should note it:

I feel like having a bunch of maps overcomplicates this a bit. Instead of spreading the data out over a bunch of different maps, we could instead have one single map for all ammo information like so:
```c++
struct AmmoInfo
{
    float damage;
    float duration;
    float cooldown;
   float respawn_time;
}

std::unordered_map<std::string, AmmoInfo> ammo_info_;
```

This would also make your parsing code quite a bit simpler too, since you wouldn't need to split out the JSON parsing across a bunch of different functions, and instead, you'd have ~2-3 small funcs:
```c++
struct AmmoInfo
{
    // ....

    // Populate our own fields from this JSON node and return if success or not
    bool ParseFromJson(const rapidjson::Value& node);
}

// For parsing the JSON file
void PickupService::LoadAmmoInformation(const Document& doc)
{
    Value::ConstMemberIterator iter = doc.MemberBegin();

    while (iter != doc.MemberEnd())
    {
        AmmoInfo info;
        info.ParseFromJson(iter->value);
        ammo_info_[iter->name] = info;
        iter++;
    }
}

```